Neos.Media/Classes/Command/MediaCommandController.php

```diff
@@ -127,6 +127,55 @@ public function importResourcesCommand($simulate = false)
     }
 
     /**
+     * Remove unused assets
+     *
+     * This command iterates over all existing assets, checks their usage count
+     * and lists the assets which are not reported as used by any AssetUsageStrategies.
+     * The unused assets can than be removed.
+     */
+    public function removeUnusedCommand()
+    {
+        $iterator = $this->assetRepository->findAllIterator();
+        $assetCount = $this->assetRepository->countAll();
+        $unusedAssetInfo = [];
+        $unusedAssetCount = 0;
+
+        $this->outputLine('<b>Searching for unused assets.</b>');
+
+        $this->output->progressStart($assetCount);
+        /** @var Asset $asset */
+        foreach ($this->assetRepository->iterate($iterator) as $asset) {
+            if ($asset->getUsageCount() === 0) {
+                $unusedAssetInfo[] = sprintf('- Found unused asset %s with file name %s', $asset->getIdentifier(), $asset->getResource()->getFilename());
+                $unusedAssetCount++;
+            }
+            $this->output->progressAdvance(1);
+        }
+
+        $this->outputLine(PHP_EOL . implode(PHP_EOL, $unusedAssetInfo));
+
+        $this->output->outputLine(sprintf('Found <b>%s</b> unused assets.', $unusedAssetCount));
+        if ($unusedAssetCount === 0) {
+            $this->quit(0);
+        }
+
+        $continue = $this->output->askConfirmation('Do you want to remove these assets?');
+        if ($continue !== true) {
+            $this->quit(0);
+        }
+
+        $iterator = $this->assetRepository->findAllIterator();
```

Comment:
instead of doing this slow lookup again, why not add the already found assets to an array or iterator and loop over those?

Reply:
Yes, please. :)